src/mrpro/data/_kdata/KData.py

```diff
@@ -200,10 +200,13 @@ def from_file(
         sort_idx = np.lexsort(acq_indices)  # torch does not have lexsort as of pytorch 2.2 (March 2024)
 
         # Finally, reshape and sort the tensors in acqinfo and acqinfo.idx, and kdata.
-        def sort_and_reshape_tensor_fields(input_tensor: torch.Tensor):
-            return rearrange(input_tensor[sort_idx], '(other k2 k1) ... -> other k2 k1 ...', k1=n_k1, k2=n_k2)
-
-        kheader.acq_info = modify_acq_info(sort_and_reshape_tensor_fields, kheader.acq_info)
+        kheader.acq_info.apply_(
+            lambda field: rearrange_acq_info_fields(
+                field[sort_idx], '(other k2 k1) ... -> other k2 k1 ...', k1=n_k1, k2=n_k2
+            )
+            if isinstance(field, torch.Tensor | Rotation)
```

Comment:
Why do we need `if isinstance` here? There is also the check in rearrange_acq_info_fields...

Reply:
Similar comment for the Mixins...

Reply:
Indexing does not work on all fields of a dataclass, only on tensors and rotation

Reply:
but if it is not a Tensor or a Rotation then rearrange_acq_info_fields simply does nothing, so why do we need to check beforehand too?

Reply:
Because the indexing in the lambda happens first

Reply:
Or we would have to move the indexing also into the rearrange_acq_info function

Reply:
Curently, it is conceptually two chained operations: indexing and reaaranging. Both should pass everything that is neither rotation nor tensor unchanged

Reply:
Once again, I think we don't need spatial dimension as we will always resources into it and only have to deal with x y z if these are tensors.

This also avoids the issue of SpatialDimension [int]

Reply:
I will take another look at these mixins, but won't spent to much time thinking about it ad I won't to make them obsolute as soon as possible anyways..

Reply:
Also more than happy to use the recurse-feature for SpatialDimension but currently it seems we are handling it differently in different Mixins. would be nice to unify it

Reply:
in SplitMixin we allow Tensor, Rotation and SpatialDimension and in SelectMixin we only allow Tensor

Reply:
Ah, got it. This should all be rotation or tensor